sklearn/tree/_criterion.pyx

```diff
@@ -1150,6 +1150,11 @@ cdef class MSE(RegressionCriterion):
         cdef intp_t k
         cdef float64_t w = 1.0
 
+        # The missing samples are assumed to be in
+        # self.sample_indices[-self.n_missing:] that is
+        # self.sample_indices[end_non_missing:self.end].
+        cdef intp_t end_non_missing = self.end - self.n_missing
```

Comment:
Let's move this part inside the `if` statement since this is only useful for missing values.

Reply:
You can also remove the comment

Reply:
We cannot define a cdef inside an if statement in Cython. Removed the comment tho.